lnbits/core/services/payments.py

```diff
@@ -791,18 +778,31 @@ async def _pay_external_invoice(
         message = payment_response.error_message or "without an error message."
         raise PaymentError(f"Payment failed: {message}", status="failed")
 
-    # payment.ok can be True (paid) or None (pending)!
-    payment.status = (
-        PaymentState.SUCCESS if payment_response.ok is True else PaymentState.PENDING
-    )
-    payment.fee = -(abs(payment_response.fee_msat or 0) + abs(service_fee_msat))
-    payment.preimage = payment_response.preimage
-    await update_payment(payment, payment_response.checking_id, conn=conn)
-    payment.checking_id = payment_response.checking_id
-    if payment.success:
+    if payment_response.success:
+        payment = await update_payment_success_status(
+            payment, payment_response, conn=conn
+        )
         await send_payment_notification(wallet, payment)
         logger.success(f"payment successful {payment_response.checking_id}")
+    elif payment.preimage is None and payment_response.preimage is not None:
+        payment.preimage = payment_response.preimage
+        await update_payment(payment, conn=conn)
```

Comment:
the only why there is a preimage in payment_response or payment is when it was successful, which is checked above